src/sinks/http.rs

```diff
@@ -148,7 +138,12 @@ impl SinkConfig for HttpSinkConfig {
             .bytes(bytesize::mib(10u64))
             .timeout(1)
             .parse_config(config.batch)?;
-        let request = config.request.tower.unwrap_with(&REQUEST_DEFAULTS);
+        let request = config.request.tower.unwrap_with(&TowerRequestConfig {
+            concurrency: Concurrency::Fixed(10),
```

Comment:
I wonder why this is fixed, generic HTTP feels like it could/would be more flexible

Reply:
It's a good question and I don't know the answer. It _was_ intentionally set this way but without comment to suggest that it's for any particular reason. Under those conditions I hesitate to adjust without more information one way or the other.

Reply:
Agreed - maybe the next approver will know 😄

Reply:
For the `http` sink, there should be no limit in my opinion. Any sink that does not align with a hosted service with documented limits should not have a limit. In those cases, the user should opt into a limit since we have no idea what is optimal.

Reply:
@binarylogic I can get behind that.

Reply:
Even then, it's not perfect, because we don't know how many Vector instances they are running, which would require dividing up these limits. This is why we built ARC. Given that, I'd be fine removing the limits altogether and relying on responses from the downstream service to backoff appropriately.

Reply:
Addressed in 50786e4.

With regard to ARC I think we can't quite remove explicit rate limit control entirely. Consider a service that blackholes requests over some published rate limit. ARC has no signal there to back off -- it might speed up! -- and we'd need some toggle to control for that. 

Overall I would like to see the tower stack we have generic for all sinks be exposed to the sink as a direct construction. Sinks that don't need rate limits, do need compression, want load balancing etc etc can construct it themselves from plug and play components we have documented. This'll get us out of the one size fits all contortion the stack here is in right now. 

That's my hunch for a path forward, anyhow. We don't have any throughput data suggesting to tackle the work presently but when we go above 10Mb/s per core this'll come back up again, I'm sure.